File: dot-net-projects-master/csasp-lms/UniversityLMS/Controllers/UserController.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Data.Entity;
using System.Web;
using System.Web.Mvc;
using UniversityLMS.DAL;
using UniversityLMS.Filters;

namespace UniversityLMS.Controllers
{
    [RAuthorizeAttribute()]
    public class UserController : System.Web.Mvc.Controller
    {
        private LMSContext db = new LMSContext();
        //
        // GET: /User/

        public ActionResult Index()
        {
            return View(db.Users.ToList());
        }

        //
        // GET: /User/Details/5

        public ActionResult Details(int id = 0)
        {
            User user = db.Users.Find(id);
            if ((user == null))
            {
                return HttpNotFound();
            }
            return View(user);
        }

        //
        // GET: /User/Create

        public ActionResult Create()
        {
            return View();
        }

        //
        // POST: /User/Create

        [HttpPost()]
        [ValidateAntiForgeryToken()]
        public ActionResult Create(User user)
        {
            if (ModelState.IsValid)
            {
                db.Users.Add(user);
                db.SaveChanges();
                return RedirectToAction("Index");
            }

            return View(user);
        }

        //
        // GET: /User/Edit/5

        public ActionResult Edit(int id = 0)
        {
            User user = db.Users.Find(id);
            if ((user == null))
            {
                return HttpNotFound();
            }
            return View(user);
        }

        //
        // POST: /User/Edit/5

        [HttpPost()]
        [ValidateAntiForgeryToken()]
        public ActionResult Edit(User user)
        {
            if (ModelState.IsValid)
            {
                db.Entry(user).State = System.Data.EntityState.Modified;
                db.SaveChanges();
                return RedirectToAction("Index");
            }

            return View(user);
        }

        //
        // GET: /User/Delete/5

        public ActionResult Delete(int id = 0)
        {
            User user = db.Users.Find(id);
            if ((user == null))
            {
                return HttpNotFound();
            }
            return View(user);
        }

        //
        // POST: /User/Delete/5

        [HttpPost()]
        [ActionName("Delete")]
        [ValidateAntiForgeryToken()]
        public RedirectToRouteResult DeleteConfirmed(int id)
        {
            User user = db.Users.Find(id);
            db.Users.Remove(user);
            db.SaveChanges();
            return RedirectToAction("Index");
        }

        protected override void Dispose(bool disposing)
        {
            db.Dispose();
            base.Dispose(disposing);
        }

    }

}

File: dot-net-projects/vbasp-project-log/VBAspProjectLog/Controllers/UserController.vb
Imports System.Data.Entity

<RAuthorizeAttribute()>
Public Class UserController
    Inherits System.Web.Mvc.Controller

    Private db As New ProjectLogContext

    '
    ' GET: /User/

    Function Index() As ActionResult
        Return View(db.Users.ToList())
    End Function

    '
    ' GET: /User/Details/5

    Function Details(Optional ByVal id As Integer = Nothing) As ActionResult
        Dim user As User = db.Users.Find(id)
        If IsNothing(user) Then
            Return HttpNotFound()
        End If
        Return View(user)
    End Function

    '
    ' GET: /User/Create

    Function Create() As ActionResult
        Return View()
    End Function

    '
    ' POST: /User/Create

    <HttpPost()> _
    <ValidateAntiForgeryToken()> _
    Function Create(ByVal user As User) As ActionResult
        If ModelState.IsValid Then
            db.Users.Add(user)
            db.SaveChanges()
            Return RedirectToAction("Index")
        End If

        Return View(user)
    End Function

    '
    ' GET: /User/Edit/5

    Function Edit(Optional ByVal id As Integer = Nothing) As ActionResult
        Dim user As User = db.Users.Find(id)
        If IsNothing(user) Then
            Return HttpNotFound()
        End If
        Return View(user)
    End Function

    '
    ' POST: /User/Edit/5

    <HttpPost()> _
    <ValidateAntiForgeryToken()> _
    Function Edit(ByVal user As User) As ActionResult
        If ModelState.IsValid Then
            db.Entry(user).State = EntityState.Modified
            db.SaveChanges()
            Return RedirectToAction("Index")
        End If

        Return View(user)
    End Function

    '
    ' GET: /User/Delete/5

    Function Delete(Optional ByVal id As Integer = Nothing) As ActionResult
        Dim user As User = db.Users.Find(id)
        If IsNothing(user) Then
            Return HttpNotFound()
        End If
        Return View(user)
    End Function

    '
    ' POST: /User/Delete/5

    <HttpPost()> _
    <ActionName("Delete")> _
    <ValidateAntiForgeryToken()> _
    Function DeleteConfirmed(ByVal id As Integer) As RedirectToRouteResult
        Dim user As User = db.Users.Find(id)
        db.Users.Remove(user)
        db.SaveChanges()
        Return RedirectToAction("Index")
    End Function

    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
        db.Dispose()
        MyBase.Dispose(disposing)
    End Sub

End Class